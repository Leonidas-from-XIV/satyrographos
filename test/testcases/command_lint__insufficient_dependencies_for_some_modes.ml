open Core
open Satyrographos_testlib

let satysfi_package_opam =
  "satysfi-package.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package"
    ~version:"0.1"
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ((package "first.satyh" "first.satyh")
            (package "second.satyg" "second.satyg")
            (package "third.satyh" "third.satyh")
           ))
  (opam "satysfi-package.opam")
  (dependencies ()))
|}

let packages = [
  "first.satyh", {|@import: second
|};
  "second.satyg", {|@import: third
|};
  "third.satyh", {|
|};
]

let opam_libs = [
]


let files =
  [ satysfi_package_opam;
    satyristes;
  ] @ packages

let () =
  TestCommand.test_lint_command ~opam_libs files
