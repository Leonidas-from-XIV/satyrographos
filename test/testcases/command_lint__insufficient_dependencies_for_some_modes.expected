@@test_library@@/second.satyg:1.1-1.9:
Error: lib/dep/missing
  Missing dependency for “@import: third” (mode (Generic))

1 problem(s) found.
